linprog: Differents solution with scipy 0.19.1 vs 1.2.1

<!-- 
Thank you for taking the time to report a SciPy issue.
Please describe the issue in detail, and for bug reports
fill in the fields below. You can delete the sections that 
don't apply to your issue. You can view the final output
by clicking the preview button above.
-->

I have a simple linear programmin problem with only equality constraints. When I try to solve it with scipy (v. 0.19.1), I get a succesful solution:
![imagen](https://user-images.githubusercontent.com/45110745/60055103-2324e680-96a2-11e9-945a-a7f1efb9d76a.png)

However, whe I try to solve the same problem (the same code below) i get an error with a linear combination in matrix A.

![imagen](https://user-images.githubusercontent.com/45110745/60055067-09839f00-96a2-11e9-8b9c-080ea49b1a65.png)


### Reproducing code example:
<!-- 
If you place your code between the triple backticks below, 
it will be marked as a code block automatically 
-->


```
import numpy as np
import scipy.optimize as opt

A_eq = np.array([[1,1,0,0,0,0],
                [0,0,1,1,0,0],
                [0,0,0,0,1,1],
                [1,0,1,0,0,0],
                [0,0,0,1,1,0],
                [0,1,0,0,0,1]])
b_eq=np.array([221,210,10,141,198,102])
c=np.concatenate((0,1,np.zeros(4)),axis=None)

sol=opt.linprog(c,A_eq=A_eq,b_eq=b_eq)
sol

```


### Scipy/Numpy/Python version information: 
Working: 
Scipy: 0.19.1 
Numpy: 1.15.4 


Not working:
Scipy: 1.2.1 
Numpy: 1.16.4 


<!-- You can simply run the following and paste the result in a code block
```
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
```
-->

Thanks a lot for the references! I was able to set everything up following the guides :) 
For testing everything passed except `TestQRdelete_F.test_delete_last_p_col` with a similar error message as #5335:

```Python
================================================================================================= FAILURES =================================================================================================
__________________________________________________________________________________ TestQRdelete_F.test_delete_last_p_col ___________________________________________________________________________________

self = <scipy.linalg.tests.test_decomp_update.TestQRdelete_F object at 0x123e020b8>

    def test_delete_last_p_col(self):
        a, q, r = self.generate('tall', 'full')
        q1, r1 = qr_delete(q, r, 0, a.shape[1], 'col')
>       assert_unitary(q1)

a          = array([[0.36852294+0.7065659j , 0.9763755 +0.8815727j ,
        0.73179656+0.5390478j , 0.6848747 +0.2577209j ,
      ...0.8069605j ,
        0.2923851 +0.9023696j , 0.08459105+0.7541054j ,
        0.8391204 +0.23152167j]], dtype=complex64)
q          = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
q1         = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
r          = array([[-3.1992414 +0.j        , -2.1899545 +0.23892963j,
        -1.6467261 +0.47293147j, -2.4033258 +0.41776413j,
  ...        ,
         0.        +0.j        ,  0.        +0.j        ,
         0.        +0.j        ]], dtype=complex64)
r1         = array([], shape=(12, 0), dtype=complex64)
self       = <scipy.linalg.tests.test_decomp_update.TestQRdelete_F object at 0x123e020b8>

scipy/linalg/tests/test_decomp_update.py:327: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
rtol = 0.0001, atol = 2.384185791015625e-07, assert_sqr = True

    def assert_unitary(a, rtol=None, atol=None, assert_sqr=True):
        if rtol is None:
            rtol = 10.0 ** -(np.finfo(a.dtype).precision-2)
        if atol is None:
            atol = 2*np.finfo(a.dtype).eps
    
        if assert_sqr:
            assert_(a.shape[0] == a.shape[1], 'unitary matrices must be square')
        aTa = np.dot(a.T.conj(), a)
>       assert_allclose(aTa, np.eye(a.shape[1]), rtol=rtol, atol=atol)
E       AssertionError: 
E       Not equal to tolerance rtol=0.0001, atol=2.38419e-07
E       
E       Mismatch: 1.39%
E       Max absolute difference: 3.57627869e-07
E       Max relative difference: inf
E        x: array([[ 1.000000e+00+0.000000e+00j, -1.676381e-08+1.490116e-08j,
E                0.000000e+00+1.490116e-08j, -2.793968e-08-2.980232e-08j,
E               -2.235174e-08-1.862645e-08j, -2.095476e-08+5.587935e-09j,...
E        y: array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
E              [0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
E              [0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.],...

a          = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
aTa        = array([[ 1.0000001e+00+0.0000000e+00j, -1.6763806e-08+1.4901161e-08j,
         0.0000000e+00+1.4901161e-08j, -2.793967...3484e-08-4.4703484e-08j,
        -8.3819032e-09-2.9802322e-08j,  9.9999994e-01+0.0000000e+00j]],
      dtype=complex64)
assert_sqr = True
atol       = 2.384185791015625e-07
rtol       = 0.0001

scipy/linalg/tests/test_decomp_update.py:21: AssertionError
```

--- a/scipy/optimize/_remove_redundancy.py
+++ b/scipy/optimize/_remove_redundancy.py
@@ -432,7 +432,7 @@ def _remove_redundancy(A, b):
                        "off redundancy removal, or try turning off presolve "
                        "altogether.")
             break
-        if np.any(np.abs(v.dot(b)) > tol * 10):  # factor of 10 to fix 10038
+        if np.any(np.abs(v.dot(b)) > tol * 100):  # factor of 100 to fix 10038 and 10349
             status = 2
             message = ("There is a linear combination of rows of A_eq that "
                        "results in zero, suggesting a redundant constraint. "

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -1312,6 +1312,24 @@ class LinprogCommonTests(object):
                       method=self.method, options=self.options)
         _assert_success(res, desired_x=[10, -3], desired_fun=-22)
 
+    def test_bug_10349(self):
+        """
+        Test for tolerance issue on LPs with equality constraints
+        https://github.com/scipy/scipy/issues/10349
+        """
+        A_eq = np.array([[1, 1, 0, 0, 0, 0],
+                         [0, 0, 1, 1, 0, 0],
+                         [0, 0, 0, 0, 1, 1],
+                         [1, 0, 1, 0, 0, 0],
+                         [0, 0, 0, 1, 1, 0],
+                         [0, 1, 0, 0, 0, 1]])
+        b_eq = np.array([221, 210, 10, 141, 198, 102])
+        c = np.concatenate((0, 1, np.zeros(4)), axis=None)
+        with suppress_warnings() as sup:
+            sup.filter(OptimizeWarning, "A_eq does not appear...")
+            res = linprog(c, A_eq=A_eq, b_eq=b_eq)
+        _assert_success(res, desired_x=[129, 92, 12, 198, 0, 10], desired_fun=92)
+
 #########################
 # Method-specific Tests #
 #########################

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -1314,7 +1314,7 @@ class LinprogCommonTests(object):
 
     def test_bug_10349(self):
         """
-        Test for tolerance issue on LPs with equality constraints
+        Test for redundancy removal tolerance issue
         https://github.com/scipy/scipy/issues/10349
         """
         A_eq = np.array([[1, 1, 0, 0, 0, 0],

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -1327,7 +1327,8 @@ class LinprogCommonTests(object):
         c = np.concatenate((0, 1, np.zeros(4)), axis=None)
         with suppress_warnings() as sup:
             sup.filter(OptimizeWarning, "A_eq does not appear...")
-            res = linprog(c, A_eq=A_eq, b_eq=b_eq)
+            res = linprog(c, A_ub, b_ub, A_eq, b_eq, bounds,
+                          method=self.method, options=self.options)
         _assert_success(res, desired_x=[129, 92, 12, 198, 0, 10], desired_fun=92)
 
 #########################

